### PR TITLE
Fixes for flake8 master

### PR DESCRIPTION
**Pull request by asottile**
I'm prepping the 3.6.0 flake8 release, figured I'd try some larger codebases :)

### CODE REVIEW

File: src/_pytest/pathlib.py [hunk 1]
```diff
@@ -245,8 +245,8 @@ def make_numbered_dir_with_cleanup(root, prefix, keep, lock_timeout):
             p = make_numbered_dir(root, prefix)
             lock_path = create_cleanup_lock(p)
             register_cleanup_lock_removal(lock_path)
-        except Exception as e:
-            pass
+        except Exception as exc:
+            e = exc
```
Comment (asottile) [inline src/_pytest/pathlib.py]: @RonnyPfannschmidt this one is subtle, and a new python3 behaviour:

```python
exc = 'hi'
try:
    raise AssertionError('hello')
except Exception as exc:
    pass
print(exc)
```

```console
$ python2 t.py
hello

$ python3 t.py 
Traceback (most recent call last):
  File "t.py", line 6, in <module>
    print(exc)
NameError: name 'exc' is not defined
```

[from the docs](https://docs.python.org/3/reference/compound_stmts.html#the-try-statement)

Comment (RonnyPfannschmidt) [reply src/_pytest/pathlib.py]: fabulous work, thanks for digging

Comment (blueyed) [reply src/_pytest/pathlib.py]: So why was the `assert e is not None` below not causing a problem before?  (since `e` should have been deleted)

Comment (blueyed) [reply src/_pytest/pathlib.py]: Ah, likely untested code - i.e. it returned instead of going there I assume.

Comment (asottile) [reply src/_pytest/pathlib.py]: Retry code that was never hit :man_shrugging:

**Pull request merged**